Automattic/woocommerce-payments
# Changes to Payment Process to store the working mode of the gateway

Fixes #7421 

#### Changes proposed in this Pull Request

- Stores the mode in `PaymentContext`
- updates the mode into `PaymentContext` in `PaymentProcessingService`
- Updates the mode into `_wcpay_mode` order metadata when the payment process finishes.

#### Testing instructions
- Place an order with the new payment process and check that `_wcpay_mode` is populated in order metadata.

-------------------

- [x] N/A - Tested on mobile (or does not apply)

**Post merge**
